src/components/welcome/MyAccounts/utils/multiChainSafe.ts

```diff
@@ -0,0 +1,128 @@
+  chainId: string
+}
+
+export const isMultiChainSafeItem = (safe: SafeItem | MultiChainSafeItem): safe is MultiChainSafeItem => {
+  if ('safes' in safe && 'address' in safe) {
+    return true
+  }
+  return false
+}
+
+const areOwnersMatching = (owners1: string[], owners2: string[]) =>
+  owners1.length === owners2.length && owners1.every((owner) => owners2.some((owner2) => sameAddress(owner, owner2)))
+
+export const getSafeSetups = (
+  safes: SafeItem[],
+  safeOverviews: SafeOverview[],
+  undeployedSafes: UndeployedSafesState,
+): (SafeSetup | undefined)[] => {
+  const safeSetups = safes.map((safeItem) => {
+    const undeployedSafe = undeployedSafes?.[safeItem.chainId]?.[safeItem.address]
+    if (undeployedSafe) {
+      const counterfactualSetup = extractCounterfactualSafeSetup(undeployedSafe, safeItem.chainId)
+      if (!counterfactualSetup) return undefined
+      return {
+        owners: counterfactualSetup.owners,
+        threshold: counterfactualSetup.threshold,
+        chainId: safeItem.chainId,
+      }
+    }
+    const foundOverview = safeOverviews?.find(
+      (overview) => overview.chainId === safeItem.chainId && sameAddress(overview.address.value, safeItem.address),
+    )
+    if (!foundOverview) return undefined
+    return {
+      owners: foundOverview.owners.map((owner) => owner.value),
+      threshold: foundOverview.threshold,
+      chainId: safeItem.chainId,
+    }
+  })
+  return safeSetups
+}
+
+export const getSharedSetup = (safeSetups: (SafeSetup | undefined)[]): Omit<SafeSetup, 'chainId'> | undefined => {
+  const comparisonSetup = safeSetups[0]
+
+  if (!comparisonSetup) return undefined
+
+  const allMatching = safeSetups.every(
+    (setup) =>
+      setup && areOwnersMatching(setup.owners, comparisonSetup.owners) && setup.threshold === comparisonSetup.threshold,
+  )
+
+  const { owners, threshold } = comparisonSetup
+  return allMatching ? { owners, threshold } : undefined
+}
+
+export const getDeviatingSetups = (
+  safeSetups: (SafeSetup | undefined)[],
+  currentChainId: string | undefined,
+): SafeSetup[] => {
+  const currentSafeSetup = safeSetups.find((setup) => setup?.chainId === currentChainId)
+  if (!currentChainId || !currentSafeSetup) return []
+
+  const deviatingSetups = safeSetups
+    .filter((setup): setup is SafeSetup => Boolean(setup))
+    .filter((setup) => {
+      return (
+        setup &&
+        (!areOwnersMatching(setup.owners, currentSafeSetup.owners) || setup.threshold !== currentSafeSetup.threshold)
+      )
+    })
+  return deviatingSetups
+}
+
+const memoizedGetProxyCreationCode = memoize(
+  async (factoryAddress: string, provider: Provider) => {
+    return Safe_proxy_factory__factory.connect(factoryAddress, provider).proxyCreationCode()
+  },
+  async (factoryAddress, provider) => `${factoryAddress}${(await provider.getNetwork()).chainId}`,
+)
+
+export const predictAddressBasedOnReplayData = async (safeCreationData: ReplayedSafeProps, provider: Provider) => {
```

Comment:
Does it make sense for this util method to be under src/components/welcome when you are also using it in serc/components/new-safe ? Maybe it should be moved to a different utils location?

Reply:
Will move it to `features/multichain/utils`